field/string.go

```diff
@@ -153,7 +153,7 @@ func (f *String) Unmarshal(v interface{}) error {
 }
 
 func (f *String) Marshal(v interface{}) error {
-	if v == nil || reflect.ValueOf(v).IsZero() {
+	if v == nil || (!reflect.ValueOf(v).CanInt() && reflect.ValueOf(v).IsZero()) {
```

Comment:
is there a test that can explain this change? I think that without `keepzero` this can stay as is.

But when we add `keepzero` we should check if value is `0` (int/*int) for `field.String`, then we should assign it.

Also, it may be similar to some other types. Not sure.

Reply:
right 
the code is exception for having int value zero